File: JANUARY/05-01-2026/1975.rb
# @param {Integer[][]} matrix
# @return {Integer}
def max_matrix_sum(matrix)
    sum = 0
    neg = 0
    mn = Float::INFINITY
    matrix.each do |row|
        row.each do |x|
            neg += 1 if x < 0
            ax = x.abs
            sum += ax
            mn = [mn, ax].min
        end
    end
    sum -= 2 * mn if neg.odd?
    sum
end

File: JANUARY/05-01-2026/1975..php
class Solution {
    function maxMatrixSum($matrix) {
        $sum = 0;
        $neg = 0;
        $mn = PHP_INT_MAX;
        foreach ($matrix as $row) {
            foreach ($row as $x) {
                if ($x < 0) $neg++;
                $ax = abs($x);
                $sum += $ax;
                if ($ax < $mn) $mn = $ax;
            }
        }
        if ($neg % 2 == 1) $sum -= 2 * $mn;
        return $sum;
    }
}